src/transport/net_ib.cc

```diff
@@ -783,6 +817,19 @@ ncclResult_t ncclIbInit(ncclDebugLogger_t logFunction, ncclProfilerCallback_t pr
     INFO(NCCL_INIT|NCCL_NET, "NET/IB : Using%s %s; OOB %s:%s", line, ncclIbRelaxedOrderingEnabled ? "[RO]" : "",
           ncclIbIfName, ncclSocketToString(&ncclIbIfAddr, addrline));
 
+    ncclIbUseInline = ncclParamIbUseInline();
+    ncclIbGdrFlushDisable = ncclParamIbGdrFlushDisable();
+
+    rcclAinic = ((rcclParamAinic() == 1) ? true : false);
+    if (rcclAinic) {
+      rcclCtsInlineData = ((rcclParamCtsInlineData() == -1) ? true : false);
+      rcclCtsOffloadEnabled = ((rcclParamCtsOffloadEnabled() == -1) ? true : false);
+      // for AINIC to disable Inline data, set env NCCL_IB_USE_INLINE=-1
+      ncclIbUseInline = ((ncclParamIbUseInline() == -1) ? false : true);
```

Comment:
rcclParamCtsInlineData et al. returns -1 if variable wasn't set.
Thus this check will ignore value of env. variable if one is set.
Should we just set default values to "1", and simplify code below?
Also, why there is a difference in NCCL_IB_USE_INLINE? It would be a hard to describe this to users that -1 is a "false" for AINIC only, but 0 is "true". and at the same time situation is opposite for anything else.

```suggestion
      rcclCtsInlineData = rcclParamCtsInlineData();
      rcclCtsOffloadEnabled = rcclParamCtsOffloadEnabled();
```

Reply:
this will be taken care of and planned to handle differently.

Reply:
refactored code per discussion on 12/11/2025 and pushed updated to PR